cfgen/cfgen

```diff
@@ -365,16 +375,18 @@ Make sure the value of data_iface in the CDF is correct."""
         total_nr_confds += nr_confds
 
         for m0d in node['m0_servers']:
-            d = m0d['io_disks']
-            assert m0d['runs_confd'] or d['data'], \
-                f"{name}: Either 'runs_confd' or 'io_disks.data' must be set"
-            assert '' not in d['data'], \
-                f"{name}: Empty strings in 'io_disks.data' are not allowed"
-            if 'meta_data' in d:
-                assert d['meta_data'], \
-                    f"{name}: 'io_disks.meta_data' must not be an empty string"
-                assert d['meta_data'] not in d['data'], \
-                    f"{name}: Meta-data disk must not belong io_disks.data"
+            for f in m0d['io_disks']:
```

Comment:
As far as I can see `m0d['io_disks']` is a dictionary with two keys: `data` and `meta_data`. Are you sure you're iterating over the the right key of the dict?

Reply:
Yes, we are iterating over multiple occurrences of `io_disks` entries and then in the next loop we iterate over elements of `io_disks` i.e. `data` and `metadata`.

Reply:
Hm.. honestly speaking it took my a while to understand what would be a result of such iteration over a dict:
```
Python 3.6.8 (default, Nov 16 2020, 16:55:22)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> t = {'first': 1, 'second': 2}
>>> for elem in t:
...     print(elem)
...
first
second
```

So long story short, `f` will be assigned to key names - i.e. it will be `data` and `meta_data`. That's not very readable, but probably fine.

The problem is that `f` variable is not used within the whole `for` loop:
```python
            for f in m0d['io_disks']:
                d = m0d['io_disks']
                assert m0d['runs_confd'] or d['data'], \
                    f"{name}: Either 'runs_confd' or" \
                    "'io_disks.data' must be set"
                assert '' not in d['data'], \
                    f"{name}: Empty strings in 'io_disks.data' are not allowed"
                if 'meta_data' in d:
                    assert d['meta_data'], f"{name}: 'io_disks.meta_data'" \
                        "must not be an empty string"
                    assert d['meta_data'] not in d['data'], \
                        f"{name}: Meta-data disk must not belong io_disks.data"
```

That's why I still suspect that there is a bug in the logic. We just twice repeat the same operation (first assign `d = m0d['io_disks']`, then run some asserts etc).

Reply:
You are correct, for loop is not required. Addressed this in the updated patch. Thank you.